internal/driver/glfw/menu_bar.go

```diff
@@ -16,11 +16,41 @@ var _ fyne.Widget = (*MenuBar)(nil)
 // MenuBar is a widget for displaying a fyne.MainMenu in a bar.
 type MenuBar struct {
 	widget.Base
-	Items []fyne.CanvasObject
-
+	Items      []fyne.CanvasObject
+	index      int
 	active     bool
-	activeItem *menuBarItem
 	canvas     fyne.Canvas
+	activeItem *menuBarItem
+}
+
+// HandleKey will process keys pressed when menu is open
+func (b *MenuBar) HandleKey(key fyne.KeyName) {
+	if key == fyne.KeyEscape {
+		b.deactivate()
+	} else if key == fyne.KeyEnter || key == fyne.KeyReturn || key == fyne.KeySpace {
+		// Simulate tapping on  menu entry
+		b.Items[b.index].(*menuBarItem).Child().Do()
+	} else if key == fyne.KeyRight {
+		// Move to the menu item at right
+		if b.index < len(b.Items)-1 {
+			b.Items[b.index].(*menuBarItem).Tapped(nil)
```

Comment:
Tapped should not be called on an item with an invalid parameter.
Would it not be suitable to call the Action()?

Reply:
No, Action() is not possible, I colud of cause duplicate the code or extract it into a new function, but I think the trick of using a nil position is fine for signaling that the event comes from the keyboard and not the mouse. I have used it several places.